Spigot-Server-Patches/0085-Allow-Reloading-of-Custom-Permissions.patch

```diff
@@ -0,0 +1,38 @@
+From 60d9d8bc4ec553833a4aab375710c41d2c9138bc Mon Sep 17 00:00:00 2001
+From: William <admin@domnian.com>
+Date: Fri, 18 Mar 2016 03:30:17 -0400
+Subject: [PATCH] Allow Reloading of Custom Permissions
+
+https://github.com/PaperMC/Paper/issues/49
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+index cb9906a..c49c022 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+@@ -819,7 +819,7 @@ public final class CraftServer implements Server {
+             try {
+                 pluginManager.addPermission(perm);
+             } catch (IllegalArgumentException ex) {
+-                getLogger().log(Level.SEVERE, "Permission in " + file + " was already defined", ex);
++                getLogger().log(Level.WARNING, "Permission in " + file + " was already defined"); // Paper - Just Display as Warning
+             }
+         }
+     }
+@@ -1833,4 +1833,14 @@ public final class CraftServer implements Server {
+     {
+         return spigot;
+     }
++
++    // Paper start
++    @Override
++    public void reloadPermissions() {
```

Comment:
This won't work. This needs to be moved to `SimplePluginManager` as:
```java
   // Paper start
    public void clearPermissions() {
        synchronized (this) {
            this.permissions.clear();
            this.defaultPerms.get(true).clear();
            this.defaultPerms.get(false).clear();
        }
    }
    // Paper end
```
or similar.

and then:
```java
    @Override
    public void reloadPermissions() {
        ((SimplePluginManager) this.pluginManager).clearPermissions();
        this.loadCustomPermissions();
    }
```
or similar.